### PR TITLE
MindSphere Mobile Native

### PR DESCRIPTION
**Pull request by StefanKellerSiemens**
@MarkvanMents : this branch is reflecting the necessary steps if a developer wants to develop a MindSphere Native app with Mendix. Would be fantastic if you could already review it, even if it is not yet released. So please do not merge right away.

Update: Target Release - end of May.

Many thanks
Stefan

### CODE REVIEW

File: content/partners/siemens/mindsphere-mobile-native.md [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+
+In this chapter you will create your own custom development app and register a deep link.
+
+1. Follow the description on how to create a [Custom Development App](https://docs.mendix.com/howto/mobile/how-to-devapps) and install it on your device or emulator as described there.
+
+1. Configure the DeepLink.
+
+    The configuration of the deep links has to be done (at the moment) on the native part for Android and iOS separately. Currently the "Native Builder App" has limited configuration possibilities for deep links so we will enable them manually.
+
+If not already done in step 2 clone your github repo locally in order to make the needed changes and checkout the **developer** branch.
```
Comment (MarkvanMents) [inline content/partners/siemens/mindsphere-mobile-native.md:132]: What step are we talking about? Should we not just give the instructions in the correct place?

File: content/partners/siemens/mindsphere-mobile-native.md [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+    ```
+
+    to install the required dependencies.
+
+* Change directory by running `cd ios` and run
+
+    ```bash
+    pod install
+    ```
+
```
Comment (MarkvanMents) [inline content/partners/siemens/mindsphere-mobile-native.md:195]: Link to cocoapods documentation?

File: content/partners/siemens/mindsphere-mobile-native.md [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+    to install the iOS dependencies. The iOS project is using CocoaPods for its dependency management. For more information on installing the CocoaPods dependency manager on your machine see CocoaPods documentation.
+
+* Open the .xcodeworkspace in the ios folder using Xcode.
```
Comment (MarkvanMents) [inline content/partners/siemens/mindsphere-mobile-native.md:198]: example command?

File: content/partners/siemens/mindsphere-mobile-native.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+          restorationHandler:restorationHandler];
+  }
+  ```
+
+* From the drop-down menu choose `dev` and the device you would like to run the app on, then click the play button to
+    start a build for your app:
+
+    ![Build ios dev app](./attachments/mindsphere-mobile-native/Xcode_build_ios_dev_app.png)
+
+    After the build succeeds the app should be starting on the selected device.
```
Comment (MarkvanMents) [inline content/partners/siemens/mindsphere-mobile-native.md:251]: Not sure what this means.

**Review by MarkvanMents — COMMENTED** [review]
Review up to section 2.4

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,8787 @@
+{
```
[the remark this answers is not in the document]
Comment (HeikoOsigus) [reply package-lock.json]: It is not needed - removed it.

File: content/partners/siemens/mindsphere-mobile-native.md [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+---
+
+# MindSphere Mobile Native
+
+## 1 Introduction
+
+This documentation describes the adoptions needed to develop native mobile apps for MindSphere with Mendix. Native mobile apps do not render inside a web view - they use native UI elements. This enables fast performance, smooth animations and allow access to all native device capabilities.
+Details about native mobile app development with Mendix can be found [here.](https://docs.mendix.com/howto/mobile/native-mobile)
+
+The **Siemens MindSphere Mobile Starter Application** is an app template which is based on the **Native Mobile Quickstart** from Mendix and includes all you need to start developing a native mobile app for your MindSphere tenant.
```
[the remark this answers is not in the document]
Comment (HeikoOsigus) [reply content/partners/siemens/mindsphere-mobile-native.md:16]: Yes as soon as it is available on the app store.

File: content/partners/siemens/mindsphere-mobile-native.md [hunk 7]
```diff
@@ -0,0 +1,478 @@
...
+
+After a successful login, your app will be restarted and a session created. Then your native home page is displayed.
+
+{{% alert type="info" %}}
+If you have trouble signing in to your application, it is a good idea to clear the cookies of your mobile browser. Please see documentation for your browser for further information.
+{{% /alert %}}
+
+### 2.6 Next steps
+
+You now have a blank application which supports authentication for your specific MindSphere application, which is an excellent starting point for feveloping your app further. We also recommend investigating the basic tutorial for MindSphere web applications [Build a MindSphere app with Mendix](https://gettingstarted.mendixcloud.com/link/path/80/Build-a-MindSphere-app-with-Mendix) to learn more about MindSphere application development in general.
```
Comment (MarkvanMents) [inline content/partners/siemens/mindsphere-mobile-native.md:376]: Do you mean "authentication for your specific MindSphere tenant"? As it is written I find it confusing.

```suggestion
You now have a blank application which supports authentication for your specific MindSphere tenant, which is an excellent starting point for feveloping your app further. We also recommend investigating the basic tutorial for MindSphere web applications [Build a MindSphere app with Mendix](https://gettingstarted.mendixcloud.com/link/path/80/Build-a-MindSphere-app-with-Mendix) to learn more about MindSphere application development in general.

```

**Review by MarkvanMents — COMMENTED** [review]
Hi @HeikoOsigus and @StefanKellerSiemens .
I've proofread the whole document now. Please can you check that I haven't changed something because I misunderstood it.
Thanks.
Mark

**Comment by MarkvanMents** [comment]
Hi Stefan (@StefanKellerSiemens ) and Heiko (@HeikoOsigus )
Do you have an expected date to publish this documentation, then I can move it into a future To Do list to plan for a final review.
Thanks.

**Comment by StefanKellerSiemens** [comment]
Hi Marc, thanks for bringing this up. As MindSphere is now able to support Deeplinks, we like to remove all the source code patching stuff out of the documentation. Documentation will be than much smaller and the whole handling will be much easier for the customer. We planned to do this next week. In four weeks from now, we like to release the stuff.
As mentioned, the changes, will be to 99% removing only. So final review should be easy. Is this working for you?

**Comment by MarkvanMents** [comment]
Great - thanks for the update. Nice if we can remove some documentation.
I'll give myself a reminder in 3 weeks to allow some time to do a final review before we publish this.

File: content/partners/siemens/mindsphere-development-considerations.md [hunk 8]
```diff
@@ -128,14 +128,33 @@ The SSO module supports two ways to get a valid MindSphere token locally. The me
...
 {{% image_container width="50%" %}}![](attachments/mindsphere-development-considerations/image19.png){{% /image_container %}}
 
 This will use the credentials you have set up under **App Credentials** in the *Authorization Management* tab of the MindSphere Developer Cockpit for this application.
 
+{{% alert type="info" %}}
+**Tip:** Use the autofill feature based on a local environment variable for the *Client Secret*.
+
+Storing the *Client Secret* inside the project is, from a security perspective, not a good idea. A better approach is to use a local environment variable. Create a user-specific environment variable with *Variable name* equal to your *Client ID* value and the *Variable value* equal to your *Client Secret* value. See step 6 below for information on how to get these values.
+
+{{% image_container width="50%" %}}![](attachments/mindsphere-development-considerations/envvariables.png){{% /image_container %}}
```
Comment (MarkvanMents) [inline content/partners/siemens/mindsphere-development-considerations.md:144]: You say not to store the secret in the app, but the image shows you storing the secret in the app. Is there a better picture, or a better explanation of setting the secret in the environment?

File: content/partners/siemens/mindsphere-development-considerations.md [hunk 9]
```diff
@@ -128,14 +128,33 @@ The SSO module supports two ways to get a valid MindSphere token locally. The me
...
+
+On startup, the system checks if there is an environment variable present with the name equal to your *Client ID* value and uses its value as *ClientSecret*.
+The *ClientID* is built from the combination of:
+
+```*Host tenant*-*Cockpit application name*-*Cockpit application version*```
+
+Ensure you have filled these constants correctly.
+If everything is setup correctly the form is auto filled and submitted.
+
+Don't forget to restart your app after you change / add the environment variable.
```
Comment (MarkvanMents) [inline content/partners/siemens/mindsphere-development-considerations.md:154]: You said restart Studio Pro. Do you mean restarting the app?

**Review by MarkvanMents — COMMENTED** [review]
Hi Stefan, @StefanKellerSiemens 
I've just gone through and proofread the last few commits. I have two questions which I've added to this review.
You can see the changes I've made in these two commits: https://github.com/mendix/docs/pull/3115/files/a914b0fefaf0f99a55b13f08f16b95d50d4561d4..6b496fa3021436bf63f609611647b0ad699cb376.
Can you quickly look at my questions and changes and confirm that I can publish this now.
Thanks.
Mark

File: content/partners/siemens/mindsphere-mobile-native.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (MarkvanMents) [reply content/partners/siemens/mindsphere-mobile-native.md]: Added link to marketplace.

File: content/partners/siemens/mindsphere-development-considerations.md (hunk 9 above)
Comment (StefanKellerSiemens) [reply content/partners/siemens/mindsphere-development-considerations.md:158]: You really have to restart Studio Pro to update the enverionment variables. That is windows behavior ;-(

Comment (MarkvanMents) [reply content/partners/siemens/mindsphere-development-considerations.md]: Reverted https://github.com/mendix/docs/pull/3115/commits/90f260b71cc17ad9519a139efed98acfbfe111ec

**Review by MarkvanMents — APPROVED** [review]

**Pull request merged**